java/build-android.gradle

```diff
@@ -30,40 +48,113 @@ android {
 	buildToolsVersion "29.0.2"
 
 	defaultConfig {
-		minSdkVersion 24
+		minSdkVersion 21
 		targetSdkVersion 28
-		versionCode 1
-		versionName "1.0"
+		versionCode = getVersionCode(project.version)
+		versionName = project.version
 
 		testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
-
 	}
 
 	android {
 		lintOptions {
 			abortOnError false
 		}
 	}
+
 	buildTypes {
 		release {
 			minifyEnabled false
+			debuggable false
 			proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
 		}
 	}
+
 	compileOptions {
-		sourceCompatibility = 1.8
-		targetCompatibility = 1.8
+		sourceCompatibility = JavaVersion.VERSION_1_8
+		targetCompatibility = JavaVersion.VERSION_1_8
 	}
 
 	sourceSets {
 		main {
 			jniLibs.srcDirs = [jniLibsDir]
 		}
 	}
+
+}
+
+task sourcesJar(type: Jar) {
+	classifier "sources"
+	from android.sourceSets.main.java.srcDirs
+}
+
+task javadoc(type: Javadoc) {
+	source = android.sourceSets.main.java.srcDirs
+	classpath += project.files(android.getBootClasspath().join(File.pathSeparator))
+}
+
+task javadocJar(type: Jar, dependsOn: javadoc) {
+	classifier = 'javadoc'
+	from javadoc.destinationDir
+}
+
+artifacts {
+	archives javadocJar
+	archives sourcesJar
 }
 
 dependencies {
 	testImplementation 'org.junit.jupiter:junit-jupiter-api:5.7.0'
 	testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.7.0'
 	testImplementation 'com.google.protobuf:protobuf-java:3.10.0'
 }
+
+publishing {
+	publications {
+		maven(MavenPublication) {
+			groupId = project.group
+			artifactId = mavenArtifactId
+			version = project.version
+
+			// Three artifacts, the `aar`, the sources and the javadoc
+			artifact("$buildDir/outputs/aar/${project.name}-release.aar")
+			artifact javadocJar
+			artifact sourcesJar
+
+			pom {
+				name = 'onnx-runtime'
+				description = 'ONNX Runtime is a performance-focused inference engine for ONNX (Open Neural Network Exchange) models.'
```

Comment:
Is mavenArtifactId the more important value and this could be 'ONNX Runtime Mobile'?

Reply:
This is the actual package name to be used for import, I don't think it allows the space in it
implementation "com.microsoft.onnxruntime:onnxruntime_mobile:1.7.0"
Maybe we can change the name to "onnx-runtime-mobile"

Reply:
onnxruntime_mobile would be more consistent with the existing packages wouldn't it?

---
In reply to: [602553085](https://github.com/microsoft/onnxruntime/pull/7138#discussion_r602553085) [](ancestors = 602553085)

Reply:
I thought PMs want the name to be onnxruntime-mobile, anyway, this is not final yet, can change before 1.8 release